fluss-server/src/test/java/com/alibaba/fluss/server/kv/rocksdb/RocksDBMetricsManagerTest.java

```diff
@@ -0,0 +1,423 @@
+
+        assertThat(manager.getCollectorCount()).isEqualTo(1);
+        assertThat(manager.isRunning()).isTrue();
+
+        collectors.add(collector);
+    }
+
+    @Test
+    void testRegisterMultipleCollectors() throws Exception {
+
+        RocksDBMetricsManager manager = RocksDBMetricsManager.getInstance();
+
+        for (int i = 0; i < 5; i++) {
+            RocksDBMetricsCollector collector = createMockCollector("test_db", "test_table", i);
+            manager.registerCollector(collector);
+            collectors.add(collector);
+        }
+
+        assertThat(manager.getCollectorCount()).isEqualTo(5);
+        assertThat(manager.isRunning()).isTrue();
+    }
+
+    @Test
+    void testUnregisterCollector() throws Exception {
+        RocksDBMetricsManager manager = RocksDBMetricsManager.getInstance();
+
+        RocksDBMetricsCollector collector = createMockCollector("test_db", "test_table", 0);
+        manager.registerCollector(collector);
+        collectors.add(collector);
+
+        assertThat(manager.getCollectorCount()).isEqualTo(1);
+
+        manager.unregisterCollector(collector);
+
+        assertThat(manager.getCollectorCount()).isEqualTo(0);
+    }
+
+    @Test
+    void testRegisterAfterShutdown() throws Exception {
+        RocksDBMetricsManager manager = RocksDBMetricsManager.getInstance();
+        manager.shutdown();
+
+        RocksDBMetricsCollector collector = createMockCollector("test_db", "test_table", 0);
+        manager.registerCollector(collector);
+
+        // Should not be registered after shutdown
+        assertThat(manager.getCollectorCount()).isEqualTo(0);
+
+        collectors.add(collector);
+    }
+
+    @Test
+    void testMetricsCollectionStartsWithFirstCollector() throws Exception {
+
+        RocksDBMetricsManager manager = RocksDBMetricsManager.getInstance();
+
+        assertThat(manager.isRunning()).isFalse();
+
+        RocksDBMetricsCollector collector = createMockCollector("test_db", "test_table", 0);
+        manager.registerCollector(collector);
+        collectors.add(collector);
+
+        assertThat(manager.isRunning()).isTrue();
+    }
+
+    @Test
+    void testBatchProcessing() throws Exception {
+        RocksDBMetricsManager manager = RocksDBMetricsManager.getInstance();
+
+        // Create many collectors to test batch processing
+        int collectorCount = 250; // More than batch size (100)
+        List<RocksDBMetricsCollector> spyCollectors = new ArrayList<>();
+
+        for (int i = 0; i < collectorCount; i++) {
+            RocksDBMetricsCollector collector = createMockCollector("test_db", "test_table", i);
+            RocksDBMetricsCollector spyCollector = spy(collector);
+            manager.registerCollector(spyCollector);
+            spyCollectors.add(spyCollector);
+            collectors.add(collector);
+        }
+
+        // Wait for metrics collection to occur
+        Thread.sleep(6000); // Wait for at least one collection cycle
```

Comment:
Do not use `Thread.sleep` for testing. This kills time and is unstable when the CPU load is high on testing machines. 

The suggested way for testing scheduler is using `ManuallyTriggeredScheduledExecutorService` in tests. See `MetricViewUpdater` and `MetricViewUpdaterTest` as an example. The key idea is adding a protected constructor for `RocksDBMetricsManager` with `ScheduledExecutorService metricsCollectionExecutor` as parameter, and create the `RocksDBMetricsManager` with the `ManuallyTriggeredScheduledExecutorService` in the test method. And manually trigger the scheduler instead of sleeping.